docs/csharp/programming-guide/generics/constraints-on-type-parameters.md

```diff
@@ -16,6 +16,7 @@ Constraints inform the compiler about the capabilities a type argument must have
 |----------------|-----------------|
 |`where T : struct`|The type argument must be a value type. Any value type except <xref:System.Nullable%601> can be specified. For more information about nullable types, see [Nullable types](../nullable-types/index.md).|
 |`where T : class`|The type argument must be a reference type. This constraint applies also to any class, interface, delegate, or array type.|
+|`where T : notnull`|The type argument must be a non nullable type. This can be a non nullable reference in C# 8 or later, or a non nullable value type. This constraint applies also to any class, interface, delegate, or array type.|
```

Comment:
```suggestion
|`where T : notnull`|The type argument must be a non-nullable type. This can be a non-nullable reference type in C# 8.0 or later, or not nullable value type. This constraint applies also to any class, interface, delegate, or array type.|
```